Firestore/core/src/firebase/firestore/model/field_value.cc

```diff
@@ -253,57 +254,71 @@ class TimestampValue : public BaseValue {
   Timestamp value_;
 };
 
+/**
+ * Represents a locally-applied Server Timestamp.
+ *
+ * Notes:
+ *   - ServerTimestampValue instances are created as the result of applying an
+ *     FSTTransformMutation (see [FSTTransformMutation applyTo]). They can only
+ *     exist in the local view of a document. Therefore they do not need to be
+ *     parsed or serialized.
+ *   - When evaluated locally (e.g. via DocumentSnapshot data), they by default
+ *     evaluate to null.
+ *   - This behavior can be configured by passing custom FieldValueOptions to
+ *     `valueWithOptions:`.
+ *   - They sort after all Timestamp values. With respect to other
+ *     ServerTimestampValues, they sort by their local_write_time.
+ */
 class ServerTimestampValue : public FieldValue::BaseValue {
  public:
-  ServerTimestampValue(Timestamp local_write_time,
-                       absl::optional<FieldValue> previous_value)
-      : local_write_time_(local_write_time),
-        previous_value_(std::move(previous_value)) {
-  }
-
-  explicit ServerTimestampValue(Timestamp local_write_time)
-      : ServerTimestampValue(local_write_time, absl::nullopt) {
+  explicit ServerTimestampValue(ServerTimestamp server_timestamp)
+      : server_timestamp_(server_timestamp) {
   }
 
   Type type() const override {
     return Type::ServerTimestamp;
   }
 
   std::string ToString() const override {
-    std::string time = local_write_time_.ToString();
+    std::string time = server_timestamp_.local_write_time().ToString();
```

Comment:
Nit: use `value()` here, for consistency with the functions below?

Reply:
Done.